### PR TITLE
Feature 2716 headers

### LINKED ISSUE #2716: Documentation: Make Headers Consistent in All MET Guides
This is a sub-issue of #*898*.
https://github.com/dtcenter/METplus/issues/898

## Describe the Sub-Issue ##
This task is for all METplus component.  This specific issue is for MET only:

The headers for the MET components User's Guides may not be consistent, but we want them to be. 

Here is the syntax from the RST docs (https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#sections):

\# with overline, for parts (e.g. the RTD documentation for the component (i.e. docs/index.rst), each guide (e.g. User's Guide, Contributor's Guide) (i.e. docs/Users_Guide/index.rst, docs/Contributors_Guide/index.rst))
\* with overline, for chapters
=, for sections
-, for subsections
^, for subsubsections
", for paragraphs

### Time Estimate ###
*Estimate the amount of work required here.*
*Issues should represent approximately 1 to 3 days of work.*

### Assignee ###
- [x] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [x] Review default **alert** labels
- [x] Select **component(s)**
- [x] Select **priority**
- [x] Select **requestor(s)**

### Milestone and Projects ###
- [x] Select **Milestone** as the next official version or **Backlog of Development Ideas**
- [x] For the next official version, select the **MET-X.Y.Z Development** project

### Funding Source ###
2783543 - Lisa has hours on this account.

### PR DESCRIPTION
**Pull request by lisagoodrich**
## Expected Differences ##

- [x] Do these changes introduce new tools, command line arguments, or configuration file options? **No**</br>
If **yes**, please describe:</br>

- [x] Do these changes modify the structure of existing or add new output data types (e.g. statistic line types or NetCDF variables)? **[No]**</br>
If **yes**, please describe:</br>

## Pull Request Testing ##

- [x] Describe testing already performed for these changes:</br>
Overall this process went smoothly.  
Most of the headers were already correct.  Many of the headers needed to be capitalized to keep our formatting consistent.  As usual, there were some exceptions. 

- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

Please double check all of the index.rst files.  I think they are correct but I'd appreciate a second set of eyes on this.
docs/index.rst
docs/Users_Guide/index.rst
docs/Contributors_Guide/index.rst
___________________

https://met.readthedocs.io/en/feature_2716_headers/Users_Guide/stat-analysis.html

I think I made the right call on these.  I'm leaving them lowercase. Per Julie, this is correct.
16.3.3.1. Job: filter
16.3.3.2. Job: summary
Job: thing_it's_doing_etc.
 
_____________________________

I didn't capitalize the questions, since they are questions.  (But they are also headers so I thought it was worth discussing.)

https://met.readthedocs.io/en/feature_2716_headers/Users_Guide/appendixA.html 
Julie: I agree that they should not be capitalized.  Thanks for asking about things that aren't clear and where there could be more than one acceptable answer.

- [x] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[No, there are no errors]**

- [x] Do these changes include sufficient testing updates? **[Yes]**

- [x] Will this PR result in changes to the test suite? **[No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

- [ ] Please complete this pull request review by **11/10/23**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR definition above.
- [ ] Ensure the PR title matches the feature or bugfix branch name.
- [ ] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)** and **Development** issue
Select: **Milestone** as the version that will include these changes
Select: **Coordinated METplus-X.Y Support** project for bugfix releases or **MET-X.Y.Z Development** project for official releases
- [ ] After submitting the PR, select the :gear: icon in the **Development** section of the right hand sidebar. Search for the issue that this PR will close and select it, if it is not already selected.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

File: docs/Contributors_Guide/dev_details/index.rst [hunk 1]
```diff
@@ -1,6 +1,6 @@
-*******************
+###################
```
Comment (jprestop) [inline docs/Contributors_Guide/dev_details/index.rst]:
```suggestion
*******************
```

Comment (jprestop) [reply docs/Contributors_Guide/dev_details/index.rst]: Development Details is at the same level as the chapter Development Environment and should have the same header.
https://met.readthedocs.io/en/feature_2716_headers/Contributors_Guide/index.html

File: docs/Contributors_Guide/dev_details/index.rst [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
 Development Details
-*******************
+###################
```
Comment (jprestop) [inline docs/Contributors_Guide/dev_details/index.rst]:
```suggestion
*******************
```

File: docs/Contributors_Guide/dev_details/tmp_file_use.rst [hunk 3]
```diff
@@ -1,7 +1,8 @@
 .. _tmp_file_use:
 
+**********************
```
Comment (jprestop) [inline docs/Contributors_Guide/dev_details/tmp_file_use.rst]:
```suggestion
```

File: docs/Contributors_Guide/dev_details/tmp_file_use.rst [hunk 4]
```diff
@@ -1,7 +1,8 @@
...
 Use of Temporary Files
-======================
+**********************
```
Comment (jprestop) [inline docs/Contributors_Guide/dev_details/tmp_file_use.rst]:
```suggestion
======================
```

File: docs/Contributors_Guide/dev_details/tmp_file_use.rst [hunk 5]
```diff
@@ -31,7 +32,7 @@ abnormally, the temporary files may remain.
 .. _tmp_files_pb2nc:
 
 PB2NC Tool
-^^^^^^^^^^
+==========
```
Comment (jprestop) [inline docs/Contributors_Guide/dev_details/tmp_file_use.rst]:
```suggestion
^^^^^^^^^^
```

File: docs/Contributors_Guide/dev_details/tmp_file_use.rst [hunk 6]
```diff
@@ -59,7 +60,7 @@ PB2NC creates the following temporary files when running:
 .. _tmp_files_point2grid:
 
 Point2Grid Tool
-^^^^^^^^^^^^^^^
+===============
```
Comment (jprestop) [inline docs/Contributors_Guide/dev_details/tmp_file_use.rst]:
```suggestion
^^^^^^^^^^^^^^^
```

File: docs/Contributors_Guide/dev_details/tmp_file_use.rst [hunk 7]
```diff
@@ -77,7 +78,7 @@ tool. Once created, it is intended to be reused in future runs.
 .. _tmp_files_bootstrap:
 
 Bootstrap Confidence Intervals
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+==============================
```
Comment (jprestop) [inline docs/Contributors_Guide/dev_details/tmp_file_use.rst]:
```suggestion
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

File: docs/Contributors_Guide/dev_details/tmp_file_use.rst [hunk 8]
```diff
@@ -106,7 +107,7 @@ Where {LINE_TYPE} is :code:`cnt`, :code:`cts`, :code:`mcts`,
 .. _tmp_files_stat_analysis:
 
 Stat-Analysis Tool
-^^^^^^^^^^^^^^^^^^
+==================
```
Comment (jprestop) [inline docs/Contributors_Guide/dev_details/tmp_file_use.rst]:
```suggestion
^^^^^^^^^^^^^^^^^^
```

File: docs/Contributors_Guide/dev_details/tmp_file_use.rst [hunk 9]
```diff
@@ -132,7 +133,7 @@ and writes the result to a temporary file.
 .. _tmp_files_python_embedding:
 
 Python Embedding
-^^^^^^^^^^^^^^^^
+================
```
Comment (jprestop) [inline docs/Contributors_Guide/dev_details/tmp_file_use.rst]:
```suggestion
^^^^^^^^^^^^^^^^
```

File: docs/Contributors_Guide/dev_details/tmp_file_use.rst [hunk 10]
```diff
@@ -154,7 +155,7 @@ temporary files.
 .. _tmp_files_tc_diag:
 
 TC-Diag Tool
-^^^^^^^^^^^^
+============
```
Comment (jprestop) [inline docs/Contributors_Guide/dev_details/tmp_file_use.rst]:
```suggestion
^^^^^^^^^^^^
```

File: docs/Users_Guide/appendixC.rst [hunk 11]
```diff
@@ -7,7 +7,7 @@ Appendix C Verification Measures
 This appendix provides specific information about the many verification statistics and measures that are computed by MET. These measures are categorized into measures for categorical (dichotomous) variables; measures for continuous variables; measures for probabilistic forecasts and measures for neighborhood methods. While the continuous, categorical, and probabilistic statistics are computed by both the Point-Stat and Grid-Stat tools, the neighborhood verification measures are only provided by the Grid-Stat tool.
 
 
-Which statistics are the same, but with different names?
+Which Statistics are the Same, but with Different Names?
```
Comment (jprestop) [inline docs/Users_Guide/appendixC.rst]:
```suggestion
Which statistics are the same, but with different names?
```

File: docs/Users_Guide/masking.rst [hunk 12]
```diff
@@ -1,17 +1,17 @@
 .. _masking:
 
 *******************************************
-Regional Verification using Spatial Masking
+Regional Verification Using Spatial Masking
```
Comment (jprestop) [inline docs/Users_Guide/masking.rst]:
```suggestion
Regional Verification using Spatial Masking
```

File: docs/index.rst [hunk 13]
```diff
@@ -145,6 +148,6 @@ To cite this documentation in publications, please refer to the MET User's Guide
    Contributors_Guide/index
 
 Index
-=====
+-----
```
Comment (jprestop) [inline docs/index.rst]:
```suggestion
=====
```
Change back to original

**Comment by jprestop** [comment]
@lisagoodrich I changed the Milestone to MET-12.0.0.

**Review by jprestop — CHANGES_REQUESTED** [review]
Hi @lisagoodrich. I have requested some changes.  When you're finished with the changes and ready for me to re-review, please click on the circular arrow icon next to my name as shown in this screenshot to let me know you're ready for a re-review.
Screen Shot 2023-11-06 at 12 05 14 PM

**Comment by lisagoodrich** [comment]
@jprestop I have made all of the changes.  I think we're good.  Do I squash and merge or would you like to do that?

**Comment by jprestop** [comment]
Hi @lisagoodrich. I haven't approved it yet, since I had requested changes. I'll approve it now and will just go ahead and squash and merge for you as I know you're on your way out the door now.

**Review by jprestop — APPROVED** [review]
@lisagoodrich Thank you for making these changes and updates.  All tests have passed. I approve this request.

**Pull request merged**